controllers/oscmachine_vm_controller.go

```diff
@@ -562,10 +576,22 @@ func reconcileDeleteVm(ctx context.Context, clusterScope *scope.ClusterScope, ma
 							machineScope.Info("Get Kcp Machine", "machineKcp", m.Name)
 							machineKcpCount++
 						}
+						if labelKey == "cluster.x-k8s.io/deployment-name" {
+							machineScope.Info("Get Kw Machine", "machineKw", m.Name)
+							machineKwCount++
+						}
+
 					}
+					machineCount = machineKwCount + machineKcpCount
 				}
 			}
+			if machineCount != 1 {
```

Comment:
what if machineCount == 0, 
if vmSpec.Replica != 1, you don't increment machineCount and machineCount = machineKwCount + machineKcpCount can still be 0,

For me, there is a use case note covered

Reply:
as machineSize > 0, so you can only have machine with cluster.x-k8s.io/control-plane or cluster.x-k8s.io/deployment-name". So your case is not possible.